app/scripts/controllers/transactions/index.js

```diff
@@ -2654,4 +2654,16 @@ export default class TransactionController extends EventEmitter {
   _getApprovalId(txMeta) {
     return String(txMeta.id);
   }
+
+  _addUnapprovedTxsApprovals() {
```

Comment:
First off, I tried calling this function on constructor, but it didn't work out. Because adding approvals are async operation in `ApprovalController` and  it couldn't be deferred before `updateBadge` call in `background.js`. 

Instead, I've added `addInitialTransactionApprovals` which actually checks and add `unapprovedTxs` to `ApprovalController` approvals. Then it calls callback, in our case it is `updateBadge`.

Reply:
As we're not awaiting the `_requestApproval` method here which in turn isn't returning the promise, won't we still be calling the `callback` immediately which would be the same as calling it after from `background.js`?